pandas/core/series.py

```diff
@@ -2132,10 +2133,15 @@ def map_f(values, f):
             else:
                 map_f = lib.map_infer
 
-        if isinstance(arg, (dict, Series)):
-            if isinstance(arg, dict):
-                arg = self._constructor(arg, index=arg.keys())
+        default_dict_types = collections.Counter, collections.defaultdict
```

Comment:
```python
issubclass(type(arg), dict) and not type(arg) is dict
```

This would evaluate as `True` for a `collections.OrderedDict` which should behave like a plain `dict`.